### ISSUE TITLE
Error on recursive pattern resolving on `cookiecutter.json` file

### ISSUE THREAD

**Issue by RaulWCosta** [cookiecutter/cookiecutter#2083]
* Cookiecutter version: 2.6.0
* Template project url: https://github.com/RaulWCosta/cookiecutter-recursive-pattern-example
* Python version: 3.8.18
* Operating System: MacOS 14.5

### Description:

I want to render the folder inside `input/` into `output/`, but I got the error:

```
Unable to render variable 'pattern_a'
Error message: 'collections.OrderedDict object' has no attribute 'pattern_b'
Context: {
    "_cookiecutter": {
        "pattern_a": "{{ cookiecutter.pattern_b }}",
        "pattern_b": "pattern_b",
        "subfolder": "subfolder"
    },
    "cookiecutter": {
        "pattern_a": "{{ cookiecutter.pattern_b }}",
        "pattern_b": "pattern_b",
        "subfolder": "subfolder"
    }
}
```

I expected the command to finish successfully with the patterns correctly substituted.

Note that if I change the positions of `pattern_a` and `pattern_b` on the `input/cookiecutter.json` file then the execution finishes successfully.

### What I've run:

Input
```
cookiecutter input -o output --no-input
```

Output
```
Unable to render variable 'pattern_a'
Error message: 'collections.OrderedDict object' has no attribute 'pattern_b'
Context: {
    "_cookiecutter": {
        "pattern_a": "{{ cookiecutter.pattern_b }}",
        "pattern_b": "pattern_b",
        "subfolder": "subfolder"
    },
    "cookiecutter": {
        "pattern_a": "{{ cookiecutter.pattern_b }}",
        "pattern_b": "pattern_b",
        "subfolder": "subfolder"
    }
}
```

**Comment by ElephantRohck** [comment]
Hi,

Can I take it up can you please tell me which is the exact file location.
I couldn't find input/cookiecutter.json file. 

Input file present in "https://github.com/RaulWCosta/cookiecutter-recursive-pattern-example" , but there is no issue that is mentioned in this link so not sure how to create a Pull Request here.

Please advise I am new to Open Source contributions.

Regards

**Comment by RaulWCosta** [comment]
Hey @ElephantRohck 
The link I added (https://github.com/RaulWCosta/cookiecutter-recursive-pattern-example) is just an example to illustrate the unexpected behaviour. You should create a Pull Request against this repo, and not mine :)

**Comment by ElephantRohck** [comment]
yes I understood that...but in the codebase I couldn't find any file containing the object :

"cookiecutter": {
        "pattern_a": "{{ cookiecutter.pattern_b }}",
        "pattern_b": "pattern_b",
        "subfolder": "subfolder"
    }

**Comment by RaulWCosta** [comment]
Oh weird, do you have access to https://github.com/RaulWCosta/cookiecutter-recursive-pattern-example/blob/main/input/cookiecutter.json?
